nav2-usability/scenario_code/lib/nested-routing/nested_routing_go_router_prototype.dart

```diff
@@ -0,0 +1,217 @@
+// Copyright 2022, the Flutter project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:go_router_prototype/go_router_prototype.dart';
+
+void main() {
+  runApp(BooksApp());
+}
+
+class BooksApp extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp.router(
+      routeInformationParser: _router.parser,
+      routerDelegate: _router.delegate,
+    );
+  }
+
+  late final _router = GoRouter(
+    routes: [
+      ShellRoute(
+        path: '/',
+        redirect: (routeState) async {
+          if (routeState.path == '/') {
+            return '/books/new';
+          }
+        },
+        builder: (context, child) {
+          late final int selectedIndex;
+          final childPath = RouteState.of(context).activeChild?.path;
+          if (childPath == null || childPath == 'books') {
+            selectedIndex = 0;
+          } else if (childPath == 'settings') {
+            selectedIndex = 1;
+          }
+
+          return AppScreen(
+            currentIndex: selectedIndex,
+            child: child,
+          );
+        },
+        routes: [
+          ShellRoute(
+            path: 'books',
+            redirect: (routeState) async {
+              if (routeState.path == '/books') {
+                return '/books/new';
+              }
+            },
+            builder: (context, child) => BooksScreen(
+              selectedTab:
+                  RouteState.of(context).activeChild!.path == 'new' ? 0 : 1,
+              child: child,
+            ),
+            routes: [
+              StackedRoute(
+                path: 'new',
+                builder: (context) => NewBooksScreen(),
+              ),
+              StackedRoute(
+                path: 'all',
+                builder: (context) => AllBooksScreen(),
+              ),
+            ],
+          ),
+          StackedRoute(
+            path: 'settings',
+            builder: (context) {
+              return SettingsScreen();
+            },
+          ),
+        ],
+      ),
+    ],
+  );
+}
+
+class AppScreen extends StatelessWidget {
+  final int currentIndex;
+  final Widget child;
+
+  const AppScreen({
+    Key? key,
+    required this.currentIndex,
+    required this.child,
+  }) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      body: AnimatedSwitcher(
+        duration: Duration(milliseconds: 400),
+        child: child,
+      ),
+      bottomNavigationBar: BottomNavigationBar(
+        currentIndex: currentIndex,
+        onTap: (idx) {
+          if (idx == 0) {
+            RouteState.of(context).goTo('/books/new');
+          } else {
+            RouteState.of(context).goTo('/settings');
```

Comment:
This line is quite long, which is unfortunate given how common `goTo` calls are. Should we have a `RouteState.goTo` method to shorten this? Something like:

```suggestion
            RouteState.goTo(context, '/settings');
```

I'm still ramping up on Flutter and its style, so please excuse me if this is a silly suggestion!